### PR TITLE
add a data migration example

### PR DESCRIPTION
**Pull request by qiaojialin**
Migrate data from one IoTDB to another IoTDB.

### CODE REVIEW

File: example/session/src/main/java/org/apache/iotdb/DataMigrationExample.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    int count = 0;
+    while (schemaIter.next()) {
+      count ++;
+      Path currentPath = new Path(schemaIter.getString("timeseries"));
+      Future future = executorService.submit(new LoadThread(count, currentPath, TSDataType.valueOf(schemaIter.getString("dataType"))));
+      futureList.add(future);
+    }
+    readerPool.closeResultSet(schemaDataSet);
+
+    for (Future future: futureList) {
```
[the remark this answers is not in the document]
Comment (qiaojialin) [reply example/session/src/main/java/org/apache/iotdb/DataMigrationExample.java:97]: fixed

File: example/session/src/main/java/org/apache/iotdb/DataMigrationExample.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+
+    schemaDataSet = readerPool
+        .executeQueryStatement("show timeseries " + path);
+    schemaIter = schemaDataSet.iterator();
+
+    List<Future> futureList = new ArrayList<>();
+    int count = 0;
+    while (schemaIter.next()) {
+      count ++;
```
[the remark this answers is not in the document]
Comment (qiaojialin) [reply example/session/src/main/java/org/apache/iotdb/DataMigrationExample.java:90]: fixed

File: example/session/src/main/java/org/apache/iotdb/DataMigrationExample.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    }
+    readerPool.closeResultSet(schemaDataSet);
+
+    for (Future future: futureList) {
+      future.get();
+    }
+    executorService.shutdown();
+
+    readerPool.close();
+    readerPool.close();
```
[the remark this answers is not in the document]
Comment (qiaojialin) [reply example/session/src/main/java/org/apache/iotdb/DataMigrationExample.java:103]: fixed

File: example/session/src/main/java/org/apache/iotdb/DataMigrationExample.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    schemaDataSet = readerPool
+        .executeQueryStatement("show timeseries " + path);
+    schemaIter = schemaDataSet.iterator();
+
+    List<Future> futureList = new ArrayList<>();
+    int count = 0;
+    while (schemaIter.next()) {
+      count ++;
+      Path currentPath = new Path(schemaIter.getString("timeseries"));
+      Future future = executorService.submit(new LoadThread(count, currentPath, TSDataType.valueOf(schemaIter.getString("dataType"))));
```
[the remark this answers is not in the document]
Comment (qiaojialin) [reply example/session/src/main/java/org/apache/iotdb/DataMigrationExample.java:92]: Yes, it's my purpose. 
awaitTermination() needs a timeout interval, which is hard to determine. We could return something using callable in the future, such as recording a log.

[the remark this answers is not in the document]
Comment (Alima777) [reply example/session/src/main/java/org/apache/iotdb/DataMigrationExample.java]: I agree with the point that if something can be reused as return value, callable will be better and it has to be.

Then as an EXAMPLE, just keep this and add some comment here to show users more choices may be better. Anyway, maybe users can decide the timeout.

**Pull request merged**